apps/judicial-system/backend/src/app/modules/aws-s3/awsS3.service.ts

```diff
@@ -2,29 +2,63 @@ import { S3 } from 'aws-sdk'
 
 import { Inject, Injectable } from '@nestjs/common'
 
+import { type Logger, LOGGER_PROVIDER } from '@island.is/logging'
 import type { ConfigType } from '@island.is/nest/config'
 
+import {
+  CaseState,
+  CaseType,
+  isCompletedCase,
+  isIndictmentCase,
+} from '@island.is/judicial-system/types'
+
 import { awsS3ModuleConfig } from './awsS3.config'
 
+const requestPrefix = 'uploads/'
+const generatedPrefix = 'generated/'
+const indictmentPrefix = 'indictments/'
+const completedIndictmentPrefix = 'indictments/completed/'
+
+const formatS3RequestKey = (key: string) => `${requestPrefix}${key}`
+const formatS3IndictmentKey = (key: string) => `${indictmentPrefix}${key}`
+const formatS3CompletedIndictmentKey = (key: string) =>
+  `${completedIndictmentPrefix}${key}`
+
+const formatS3Key = (caseType: CaseType, caseState: CaseState, key: string) =>
+  isIndictmentCase(caseType)
+    ? isCompletedCase(caseState)
+      ? formatS3CompletedIndictmentKey(key)
+      : formatS3IndictmentKey(key)
+    : formatS3RequestKey(key)
+
+const formatConfirmedKey = (key: string) =>
```

Comment:
Super nitpick -- Consider moving this function to line nr 26, so that all the `format` functions are in the same place.